account_asset_management/models/account_asset.py

```diff
@@ -495,15 +474,16 @@ def name_search(self, name, args=None, operator="ilike", limit=100):
             if operator in expression.NEGATIVE_TERM_OPERATORS:
                 domain = ["&", "!"] + domain[1:]
         assets = self.search(domain + args, limit=limit)
-        return assets.name_get()
+        return assets._compute_display_name()
 
     @api.depends("name", "code")
-    def name_get(self):
+    def _compute_display_name(self):
         result = []
         for asset in self:
             name = asset.name
             if asset.code:
                 name = " - ".join([asset.code, name])
+            asset.display_name = name
             result.append((asset.id, name))
         return result
```

Comment:
A computed method shouldn't return anything. In fact, the computed method assign the value through dot notation.